### PR TITLE
#525 no longer send rejection mails to non-existing people

### PR DESCRIPTION
**Pull request by ekrebs5**
#525

### CODE REVIEW

File: spec/features/email_spec.rb [hunk 1]
```diff
@@ -40,6 +40,20 @@
     expect(Event.find(@event.id).acceptances_have_been_sent).to be(true)
   end
 
+  scenario "logged in as Organizer after sending an acceptance Email to the applicants of an event which has no rejected applications the event application status for sending rejections gets locked" do
+    @event = FactoryGirl.create(:event_with_accepted_applications,
+                                rejected_application_letters_count: 0,
```
Comment (aloeser) [inline spec/features/email_spec.rb:45]: you could use :in_selection_phase here, not sure if we really need it tho

File: app/models/event.rb [hunk 2]
```diff
@@ -193,6 +193,14 @@ def compute_occupied_places
     application_letters.where(status: ApplicationLetter.statuses[:accepted]).count
   end
 
+  # Returns the number of currently rejected applications with no mail sent of the event
+  #
+  # @param none
+  # @return [Int] for number of rejected applications
+  def compute_number_of_rejections_without_mail
```
Comment (aloeser) [inline app/models/event.rb]: not wrong, but I think what we want to check is if there are rejected participants at all.
maybe we should rename the function to has_rejected_participants and perform the > 0 check in here?

File: spec/features/email_spec.rb (hunk 1 above)
Comment (ekrebs5) [reply spec/features/email_spec.rb:45]: the other test don't have it
(same reasoning, phase those not effect email sending once you are on the email site)

unless we find a convention for this (see the discussion alex started on our slack), I would leave it like those

File: app/models/event.rb [hunk 3]
```diff
@@ -193,6 +193,14 @@ def compute_occupied_places
     application_letters.where(status: ApplicationLetter.statuses[:accepted]).count
   end
 
+  # Returns whether there are rejected applications with no status notification sent of the event
+  #
+  # @param none
+  # @return [Boolean] true if there are rejected participants without status notification sent
+  def has_rejected_participants_without_status_notification?
+    application_letters.where(status: ApplicationLetter.statuses[:rejected], status_notification_sent: false).count > 0
```
Comment (querenker) [inline app/models/event.rb]: I think you can use [exists?](http://apidock.com/rails/ActiveRecord/Base/exists%3F/class). I think it is a little bit better to read and with many applications_letters it is lots faster.

**Review by querenker — CHANGES_REQUESTED** [review]
Otherwise it looks good to me

**Review by querenker — APPROVED** [review]

**Review by aloeser — APPROVED** [review]
worked for me

**Pull request merged**